common/server.cpp

```diff
@@ -84,16 +84,28 @@ AuthenticationResult Server::loginUser(Server_ProtocolHandler *session, QString
         name = name.left(35);
 
     Server_DatabaseInterface *databaseInterface = getDatabaseInterface();
-
+    
     AuthenticationResult authState = databaseInterface->checkUserPassword(session, name, password, clientid, reasonStr, secondsLeft);
     if (authState == NotLoggedIn || authState == UserIsBanned || authState == UsernameInvalid || authState == UserIsInactive)
         return authState;
 
     ServerInfo_User data = databaseInterface->getUserData(name, true);
     data.set_address(session->getAddress().toStdString());
     name = QString::fromStdString(data.name()); // Compensate for case indifference
-
+    
     if (authState == PasswordRight) {
+        //check to see if user is full but allow users that donate access to server regardless of user count
+        if (!databaseInterface->isUserADonator(name)) {
```

Comment:
In the code I'd prefer to keep it agnostic - `isUserAllowedToBypassUserCountLimit` or something. I know it's verbose but I don't think the game server should care about donations. We can leave it to each server operator to determine how users are granted this bypass permission.